Option to use fake data producers

floriangroetschla opened the pull request:
This adds an option to use fake data producers that always answer with empty fragments upon request. One fake data producer replaces the (fake) card reader and DLH combination. The fake data producer module can not be used with software tpg or the DQM in its current state. The latter is because it would require adding more queues to it.

bieryAtFnal reviewed (approved):
I ran a couple of tests with FakeDataProducers enabled, and things looked fine.  I was moderately surprised that the Fragment sizes were the same as with FakeCardReaders (given the same readout window size).  If that is *not* supposed to be the case, I should look closer.

I also ran some regression tests with the 'normal' Readout modules enabled and SWTPG enabled.  Those tests ran normally.

floriangroetschla commented:
The faking of the correct fragment sizes (at least for fixed rate and fixed size frontends) is a feature I added to the fake data producer. There are two configurable parameters for this: the `time_tick_diff` (between frames) and `frame_size`. When receiving a request, the fake producer can use them to compute the number of expected frames and bytes.
